### PR TITLE
DDF-4106 Migrates workspace menus to react

### PR DESCRIPTION
**Pull request by andrewkfiedler**
#### What does this PR do?
Migrates workspace menus to react.  Adds common portal component,  as well as dropdown component.

#### Who is reviewing it? 
@codice/ui 
@djblue 
@rzwiefel 
@AzGoalie 
@samuelechu 
@adimka 

#### How should this be tested?
Verify the workspace menu works as it did previously.  I've added screenshots of the components in question that should be tested to verify functional parity.

#### What are the relevant tickets?
[DDF-4106](https://codice.atlassian.net/browse/DDF-4106)

screen shot 2018-10-30 at 8 28 27 am
screen shot 2018-10-30 at 8 28 20 am
screen shot 2018-10-30 at 8 28 15 am

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/workspace-interactions/workspace-interactions.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
+/**
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/workspace-interactions/workspace-interactions.tsx]: I think this is a container component

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/button-behavior/button-behavior.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
+/**
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/button-behavior/button-behavior.tsx]: ❓ this doesn't seem like a presentational component, maybe you can split the business logic into a container component

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/button/button.tsx [hunk 3]
```diff
@@ -106,13 +106,14 @@ type RootProps = {
   inText?: boolean
   buttonType: buttonTypeEnum
   fadeUntilHover?: boolean
+  gaseous: boolean
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/button/button.tsx]: Seems like this file would be good to split up as well (presentational/container)

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/workspace-title/workspace-title.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+type Props = {
+  title: string
+  saved: boolean
+  onChange: (value: string) => void
+}
+type State = {}
+
+class WorkspaceTitle extends React.Component<Props, State> {
+  constructor(props: Props) {
```
[the remark this answers is not in the document]
Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/container/workspace-title/workspace-title.tsx]: This file is deleted now.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/workspace-title/workspace-title.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    super(props)
+  }
+  render() {
+    return (
+      <WorkspaceTitlePresentation
+        title={this.props.title}
```
[the remark this answers is not in the document]
Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/container/workspace-title/workspace-title.tsx]: This file is deleted now.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/workspace-title/workspace-title.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import { hot } from 'react-hot-loader'
+
+type Props = {
+  title: string
+  saved: boolean
+  onChange: (value: string) => void
+}
+type State = {}
+
+class WorkspaceTitle extends React.Component<Props, State> {
```
[the remark this answers is not in the document]
Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/container/workspace-title/workspace-title.tsx]: All the container logic went away in the end, so now it's deleted.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/workspace-interactions/workspace-interactions.tsx [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+}
+
+const mapPropsToState = (props: Props) => {
+  return {
+    subscribed: props.workspace.get('subscribed'),
+  }
+}
+
+class WorkspaceInteractions extends React.Component<Props, State> {
+  constructor(props: Props) {
```
[the remark this answers is not in the document]
Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/container/workspace-interactions/workspace-interactions.tsx]: I'm not sure we can do this given we need the props to set the initial state.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/workspaces-items-container/workspaces-items-container.tsx [hunk 8]
```diff
@@ -141,19 +148,66 @@ class WorkspacesItemsContainer extends React.Component<
...
+    })
+  }
   save(key: string) {
     return function(_unused_model: any, value: any) {
       var prefs = user.get('user').get('preferences')
       prefs.set(key, value[0])
       prefs.savePreferences()
     }
   }
+  save2(key: string, value: string) {
```
[the remark this answers is not in the document]
Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/container/workspaces-items-container/workspaces-items-container.tsx]: I haven't converted both dropdowns to react yet, so I decided to keep this one around but name it reactComponentSave.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/dropdown/dropdown.tsx [hunk 9]
```diff
@@ -0,0 +1,368 @@
+/**
```
[the remark this answers is not in the document]
Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/dropdown/dropdown.tsx]: What part of it looks like business logic at the moment?

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/workspace-interactions/workspace-interactions.tsx (hunk 1 above)
Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/workspace-interactions/workspace-interactions.tsx]: Why so?

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/button-behavior/button-behavior.tsx (hunk 2 above)
Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/button-behavior/button-behavior.tsx]: Why so?

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/button/button.tsx (hunk 3 above)
Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/button/button.tsx]: Why so?

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/theme-container/theme-container.tsx [hunk 10]
```diff
@@ -222,6 +222,7 @@ const sharedState: ThemeInterface = {
   screenBelow: (specifiedSize: string) => {
     return sharedState.screenSize < parseFloat(specifiedSize)
   },
+  currentBackground: 'black',
```
[the remark this answers is not in the document]
Comment (adimka) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/container/theme-container/theme-container.tsx]: This is just a default value.

**Comment by andrewkfiedler** [comment]
build now

**Pull request merged**